x-pack/plugins/watcher/public/sections/watch_edit/components/watch_visualization.tsx

```diff
@@ -0,0 +1,177 @@
+import { VisualizeOptions } from 'plugins/watcher/models/visualize_options';
+import { getWatchVisualizationData } from '../../../lib/api';
+import { WatchContext } from './watch_context';
+import { aggTypes } from '../../../models/watch/agg_types';
+
+const getTimezone = () => {
+  const config = chrome.getUiSettingsClient();
+  const DATE_FORMAT_CONFIG_KEY = 'dateFormat:tz';
+  const isCustomTimezone = !config.isDefault(DATE_FORMAT_CONFIG_KEY);
+  if (isCustomTimezone) {
+    return config.get(DATE_FORMAT_CONFIG_KEY);
+  }
+
+  const detectedTimezone = moment.tz.guess();
+  if (detectedTimezone) {
+    return detectedTimezone;
+  }
+
+  const tzOffset = moment().format('Z');
+  return tzOffset;
+};
+
+const getDomain = (watch: any) => {
+  const VISUALIZE_TIME_WINDOW_MULTIPLIER = 5;
+  const fromExpression = `now-${watch.timeWindowSize * VISUALIZE_TIME_WINDOW_MULTIPLIER}${
+    watch.timeWindowUnit
+  }`;
+  const toExpression = 'now';
+  const fromMoment = dateMath.parse(fromExpression);
+  const toMoment = dateMath.parse(toExpression);
+  const visualizeTimeWindowFrom = fromMoment ? fromMoment.valueOf() : 0;
+  const visualizeTimeWindowTo = toMoment ? toMoment.valueOf() : 0;
+  return {
+    min: visualizeTimeWindowFrom,
+    max: visualizeTimeWindowTo,
+  };
+};
+
+const getTimeBuckets = (watch: any) => {
+  const domain = getDomain(watch);
+  const timeBuckets = new TimeBuckets();
+  timeBuckets.setBounds(domain);
+  return timeBuckets;
+};
+
+const loadWatchVisualizationData = debounce(async (watch: any, setWatchVisualizationData: any) => {
+  const domain = getDomain(watch);
+  const timeBuckets = new TimeBuckets();
+  timeBuckets.setBounds(domain);
+  const interval = timeBuckets.getInterval().expression;
+  const visualizeOptions = new VisualizeOptions({
+    rangeFrom: domain.min,
+    rangeTo: domain.max,
+    interval,
+    timezone: getTimezone(),
+  });
+  const { visualizeData } = (await getWatchVisualizationData(watch, visualizeOptions)) as any;
+  const agg = aggTypes[watch.aggType].value;
+  const data = visualizeData[agg];
+  setWatchVisualizationData(data || []);
+}, 500);
+
+const WatchVisualizationUi = () => {
+  const { watch } = useContext(WatchContext);
+  const [watchVisualizationData, setWatchVisualizationData] = useState<number[][]>([]);
+
+  useEffect(
+    () => {
+      loadWatchVisualizationData(watch, setWatchVisualizationData);
+    },
+    [watch]
+  );
+
+  const timezone = getTimezone();
+  const dateFormatter = (d: number) => {
+    return moment(d)
+      .tz(timezone)
+      .format(getTimeBuckets(watch).getScaledDateFormat());
+  };
+  const aggLabel = aggTypes[watch.aggType].text;
+  const thresholdCustomSeriesColors: CustomSeriesColorsMap = new Map();
+  const thresholdDataSeriesColorValues: DataSeriesColorsValues = {
+    colorValues: [],
+    specId: getSpecId('threshold'),
+  };
+  thresholdCustomSeriesColors.set(thresholdDataSeriesColorValues, 'red');
```

Comment:
nit: should we use the hex from the `eui` color pallete? it's `#BD271E`

https://elastic.github.io/eui/#/guidelines/colors

Reply:
makes sense, done